src/test/resources/application.properties

```
@@ -0,0 +1,11 @@
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driver-class-name=org.h2.Driver 
```

Тут у тебя лишний пробел
```suggestion
spring.datasource.driver-class-name=org.h2.Driver
```